src/platform/PlatformLockSupport.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    Copyright (c) 2018 Nest Labs, Inc.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Provides implementations of the CHIP System Layer platform
+ *          event functions that are suitable for use on all platforms.
+ */
+/* this file behaves like a config.h, comes first */
+#include <platform/internal/CHIPDeviceLayerInternal.h>
+
+#include <platform/PlatformManager.h>
+#include <system/PlatformLockSupport.h>
+
+namespace chip {
+namespace System {
+
+using namespace ::chip::DeviceLayer;
+
+#if !CHIP_SYSTEM_CONFIG_NO_LOCKING
```

Comment:
please make these decisions in build.gn: include the CPP file only if platform locking exists.

Depending on usage, we may also consider creating a separate target of `locking-not-available-do-not-use` (or some other name) that contains these files in case of locking not available, so that GN fails to complie if people try to use locking when no locking available.

Reply:
Removed the macro in this file and this file will be include when `chip_system_config_locking != "none"` in gn